Owncloud iOs Client/ExternalLibreries/ELCPickerLib/ELCAssetCell.m

```diff
@@ -7,84 +7,105 @@
-	if(self) {
+    self = [super initWithStyle:UITableViewCellStyleDefault reuseIdentifier:reuseIdentifier];
+	if (self) {
         UITapGestureRecognizer *tapRecognizer = [[UITapGestureRecognizer alloc] initWithTarget:self action:@selector(cellTapped:)];
         [self addGestureRecognizer:tapRecognizer];
-        [tapRecognizer release];
         
         NSMutableArray *mutableArray = [[NSMutableArray alloc] initWithCapacity:4];
         self.imageViewArray = mutableArray;
-        [mutableArray release];
         
         NSMutableArray *overlayArray = [[NSMutableArray alloc] initWithCapacity:4];
         self.overlayViewArray = overlayArray;
-        [overlayArray release];
-
-        [self setAssets:assets];
+        
+        self.alignmentLeft = YES;
+        self.imageManager = [[PHCachingImageManager alloc] init];
 	}
 	return self;
 }
 
 - (void)setAssets:(NSArray *)assets
 {
     self.rowAssets = assets;
-	for (UIView *view in [self subviews]) {
-        if ([view isKindOfClass:UIImageView.class]) {
-            [view removeFromSuperview];
-        }
+	for (UIImageView *view in _imageViewArray) {
+        [view removeFromSuperview];
+	}
+    for (ELCOverlayImageView *view in _overlayViewArray) {
+        [view removeFromSuperview];
 	}
-    
     //set up a pointer here so we don't keep calling [UIImage imageNamed:] if creating overlays
+    
+    
     UIImage *overlayImage = nil;
     for (int i = 0; i < [_rowAssets count]; ++i) {
-
+        
         ELCAsset *asset = [_rowAssets objectAtIndex:i];
         
+        PHImageRequestOptions *options = [[PHImageRequestOptions alloc] init];
+        
+        // Download from cloud if necessary
+        // Need to make NO for existing images.
+        options.networkAccessAllowed = YES;
+        options.progressHandler = ^(double progress, NSError *error, BOOL *stop, NSDictionary *info) {
+            
+        };
+        
         if (i < [_imageViewArray count]) {
             UIImageView *imageView = [_imageViewArray objectAtIndex:i];
-            imageView.image = [UIImage imageWithCGImage:asset.asset.thumbnail];
+            PHAsset *phAsset = (PHAsset *)asset.asset;
+            [self.imageManager requestImageForAsset:phAsset targetSize:CGSizeMake(70, 70) contentMode:PHImageContentModeAspectFill options:options resultHandler:^(UIImage * result, NSDictionary * info) {
+                imageView.image = result;
+            }];
+            
         } else {
-            UIImageView *imageView = [[UIImageView alloc] initWithImage:[UIImage imageWithCGImage:asset.asset.thumbnail]];
+            UIImageView *imageView = [[UIImageView alloc] initWithFrame:CGRectMake(0, 0, 70, 70)];
+            PHAsset *phAsset = (PHAsset *)asset.asset;
+            [self.imageManager requestImageForAsset:phAsset targetSize:CGSizeMake(70, 70) contentMode:PHImageContentModeAspectFill options:options resultHandler:^(UIImage * result, NSDictionary * info) {
```

Comment:
This code is also in the else structure.

 [self.imageManager requestImageForAsset:phAsset targetSize:CGSizeMake(70, 70) contentMode:PHImageContentModeAspectFill options:options resultHandler:^(UIImage * result, NSDictionary * info) {                imageView.image = result;
            }];